src/c++/vdo/base/vio.c

```diff
@@ -207,22 +216,21 @@ int vio_reset_bio(struct vio *vio, char *data, bio_end_io_t callback,
 	bio->bi_ioprio = 0;
 	bio->bi_io_vec = bio->bi_inline_vecs;
 	bio->bi_max_vecs = vio->block_count + 1;
-	len = VDO_BLOCK_SIZE * vio->block_count;
+	if (VDO_ASSERT(size <= vio_size, "specified size %d must be larger than allocated %d",
+		       size, vio_size))
```

Comment:
Please spell this like our other assertions. VDO_ASSERT() returns a result, we assuming 0 == success is techincally not correct.
Also, please fix the message. assertion message are phrased so they should describe the desired condition. Something like "specified vio size %d is not greater that allocated vio size %d".

You could potentially use VDO_ASSERT_LOG_ONLY() instead; having a size too large I think just means that bio_add_page() will fail farther down. Though by that argument perhaps we shouldn't even check this value, since it'll get caught later on.

Reply:
fixing message wording, comparing against VDO_SUCCESS